lib/Doctrine/Persistence/Mapping/ProxyClassNameResolver.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Doctrine\Persistence\Mapping;
+
+interface ProxyClassNameResolver
+{
+    public function resolveClassName(string $className): string;
```

Comment:
Do any of these need to be documented as `class-string`? Or are they not really/not always class strings?

Reply:
Yep, could be documented as class-string. Does this look correct?

```suggestion
    /** @psalm-var class-string $className */
    public function resolveClassName(string $className): string;
```

Reply:
That should be `psalm-param`, should the same be done for `psalm-return`?

Reply:
I'll be honest: I never used Psalm before, so I'm not sure what's required. I've updated the comment above and will update all necessary code points if this looks ok.

Reply:
It looks perfect!